core/src/test/java/org/elasticsearch/http/netty/NettyHttpChannelTests.java

```diff
@@ -414,5 +460,68 @@ public BytesReference content() {
         public RestStatus status() {
             return RestStatus.OK;
         }
+
+    }
+
+    private static class TestCircuitBreaker implements CircuitBreaker {
```

Comment:
I don't think we need any concurrency guarantees here as the test is single-threaded.